optimum/fx/parallelization/distributed/dist_ops.py

```diff
@@ -0,0 +1,113 @@
+import torch
+import torch.distributed as dist
+
+def all_reduce(group: dist.ProcessGroup, tensor : torch.Tensor) -> torch.Tensor:
+    world_size = dist.get_world_size(group)
+    if world_size == 1:
+        return tensor
+    
+    dist.all_reduce(tensor, group=group)
+    return tensor
+
+def all_gather(group: dist.ProcessGroup, tensor: torch.Tensor, gather_dim: int = -1) -> torch.Tensor:
+    world_size = dist.get_world_size(group)
+    if world_size == 1:
+        return tensor
+    rank = dist.get_rank(group = group)
+
+    tensor = tensor.contiguous()
+    tensors = [torch.empty_like(tensor) for _ in range(world_size)]
+    tensors[rank] = tensor
+    
+    dist.all_gather(tensors, tensor, group=group)
+    return torch.cat(tensors, dim=gather_dim)
+
+def split(group: dist.ProcessGroup, tensor: torch.Tensor, split_dim: int = -1) -> torch.Tensor:
+    world_size = dist.get_world_size(group)
+    if world_size == 1:
+        return tensor
+
+    rank = dist.get_rank(group)
+    size = tensor.size()
+    assert size[split_dim] % world_size == 0
+    tensors = torch.split(tensor, size[split_dim] // world_size, dim = split_dim)
+    tensor = tensors[rank].contiguous()
```

Comment:
why contiguous?

Reply:
tensors after `split` may not be contiguous, I think it's better be contiguous